Cargo.toml

```diff
@@ -216,6 +216,7 @@ tokio-postgres = { version = "0.5.5", optional = true, features = ["runtime", "w
 thread_local = "= 1.0.1"
 dashmap = "3"
 encoding_rs = { version = "0.8", features = ["serde"] }
+parking_lot = "0.11.1"
```

Comment:
Using parking_lot here for its `RwLock` implementation, which according to the [readme](https://github.com/Amanieu/parking_lot):

> The numbers for RwLock vary depending on the number of reader and writer threads, but are almost always faster than the standard library RwLock, and even up to 50x faster in some cases.

Given this claim, I was surprised it wasn't being used in other places where we've reached for the stdlib version. There may be design decisions/considerations I'm not privy to that preclude its use - please advise if this sticks out as being undesirable.

If this holds up as a decent choice, there's further usage in API code I could swap it out for.